src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,90 @@
+        System.out.println("todo {task description}                         Add ToDo into list");
+        System.out.println("deadline {task description} /by {DATE}          Add Deadline into list");
+        System.out.println("event {task description} /at {DATE}             Add Event into list");
+        System.out.println("mark {Task ID}                                  Mark specific task as done");
+        System.out.println("unmark {Task ID}                                Mark specific task as not done");
+        System.out.println("find {keyword}                                  Find tasks with a specific keyword");
+        System.out.println("delete {Task ID}                                Delete specific task from list");
+        System.out.println("bye                                             End the Duke program");
+    }
+
+    /**
+     * Prints out a list of available commands in this program.
+     *
+     * @param list an arraylist containing the different tasks.
+     */
+    public static void allowUserInput(TaskList list) throws DukeException, IOException {
+        // Scanner Object
+        Scanner scanner = new Scanner(System.in);
+        String input = scanner.nextLine();
+
+        while (!input.equals("bye")) {
+            String[] temp = input.split(" ");
+            Parser parse = new Parser(input);
+            if (parse.getCmd().equals("?")) {
+                getHelp();
+            } else if (parse.getCmd().equals("mark")) {
+                int index = Integer.parseInt(parse.getDesc()) - 1;
+                list.markTask(index);
+            } else if (parse.getCmd().equals("unmark")) {
+                int index = Integer.parseInt(parse.getDesc()) - 1;
+                list.unMarkTask(index);
+            } else if (parse.getCmd().equals("list")) {
+                list.listTask();
+            } else if (parse.getCmd().equals("todo")) {
+                list.addToDo(parse.getDesc(), false);
+            } else if (parse.getCmd().equals("deadline")) {
+                list.addDeadline(parse.getDesc(), parse.getDate(), false);
+            } else if (parse.getCmd().equals("event")) {
+                list.addEvent(parse.getDesc(), parse.getDate(), false);
+            } else if (parse.getCmd().equals("delete")) {
+                int index = Integer.parseInt(parse.getDesc()) - 1;
+                list.deleteTask(index);
+            } else if (parse.getCmd().equals("find")) {
+                list.findTask(parse.getDesc());
+            } else {
+                notSpecified();
+            }
+            input = scanner.nextLine();
+        }
+        // Close program and write back to file
+        Storage.writeFile(list);
+        System.out.println("Bye. Hope to see you again soon!");
+    }
+
+    /**
+     * Handles invalid inputs.
+     *
+     * @throws DukeException if invalid inputs are detected.
+     */
+    public static void notSpecified() throws DukeException {
```

Comment:
Our coding standard says that "Names representing methods must be verbs and written in camelCase." Perhaps you can consider changing the name of `notSpecified()` to fit the conventions better?